# By replacing the 1st digit of the 2-digit number $*3$, it turns out that six of the nine possible values: 13, 23, 43, 53, 73, and 83, are all prime.

# By replacing the 3rd and 4th digits of $56**3$ with the same digit, this 5-digit number is the first example having seven primes among the ten generated numbers, yielding the family: 56003, 56113, 56333, 56443, 56663, 56773, and 56993. Consequently 56003, being the first member of this family, is the smallest prime with this property.

# Find the smallest prime which, by replacing part of the number (not necessarily adjacent digits) with the same digit, is part of an eight prime value family.

_____

- First, we define a function to check for primality

In [7]:
def is_prime(n):
    max_val = (n**0.5)//1 + 1
    for i in range(2, int(max_val)):
        if n%i == 0:
            return False
    return True

- Now we define a function for replacing digits

In [59]:
def replacements(n, list_digits_to_replace):
    # list_digits to replace specifies which indices in range(len(n)) will be replaced with a *
    s = str(n)
    for i in list_digits_to_replace:
        s = s[:i]+'*'+s[i+1:]
    if 0 in list_digits_to_replace:
        list_replacements = [int(s.replace('*',str(i))) for i in range(1,10)]
    else:
        list_replacements = [int(s.replace('*',str(i))) for i in range(10)]
    return list_replacements

In [60]:
from itertools import combinations

In [63]:
i = 100
found = False

while not found:
    i += 1
    if is_prime(i):
        n_replacements = range(1, len(str(i)))
        string = ''.join([str(x) for x in range(len(str(i+1)))])
        for m in n_replacements:
            list_replacement_combinations = [x for x in combinations(string, m)]
            for combination in list_replacement_combinations:
                list_digits_to_replace = [int(x) for x in combination]
                list_replacements = replacements(i, list_digits_to_replace)
                if len([x for x in list_replacements if is_prime(x)])==8:
                    print(i, list_digits_to_replace, list_replacements)
                    found = True

120383 [0, 2, 4] [121313, 222323, 323333, 424343, 525353, 626363, 727373, 828383, 929393]


# Answer: 121313

### Note: I don't really get the question, since 120383 is a prime and replacing some of the digits creates an 8 prime family, but whatever